replaceholder.js

```diff
@@ -1,23 +1,64 @@
-(function ($) {
-    $.fn.replaceholder = function () {
+(function (global, factory) {
+    'use strict';
+
+    if (typeof define === 'function' && define.amd) {
+        // AMD require.js
+        define('replaceholder', ['jquery'], function ($) {
+            $.fn.replaceholder = factory();
+        });
+    } else if (typeof exports !== 'undefined') {
+        // Node.js CommonJS
+        require('jquery').fn.replaceholder = factory();
+    } else if (typeof global.$ === 'function' && typeof global.$.fn === 'object') {
+        // Global scope
+        global.$.fn.replaceholder = factory();
+    }
+}(this, function () {
+    'use strict';
+
+    // options argument is an object that may contain:
+    // src: 'newImage.png' // required image url
```

Comment:
I didn't mean this is required, I meant Image url that user wants to load (it's required image for this element). I'll change it to make it more clear.